tools-local/doc.tasks/GenerateMarkdownDoc.cs

```diff
@@ -94,15 +96,31 @@ private void WriteSectionProjecs(ITaskItem section, StreamWriter file)
             file.WriteLine($"Projects in section: {projectFiles.Length}\n");
                     
             Array.Sort(projectFiles, (a, b) => a.CompareTo(b));
+            Array.ForEach(projectFiles, pf => pf = RelativizePath(pf));
 
-            for (int i = 0; i < projectFiles.Length; i++)
+            var groupedPaths = projectFiles
+                .Where(pf => pf.StartsWith(RepositoryRoot))
+                .Select(pf => pf.Substring(pf.IndexOf(RepositoryRoot) + RepositoryRoot.Length))
+                .Select(pf => pf.Split('\\'))
+                .ToLookup(pf => $"{pf[0]}\\{pf[1]}", pf => new Tuple<string, string>(pf[2], pf[3]));
+
+            foreach (var srcFolders in groupedPaths)
             {
-                var projectPath = projectFiles[i];
-                var desc = GetDescriptions(projectPath);
-                var link = GenerateRelativeLink(projectPath);
-                var projectBullet = Path.GetFileName(projectPath);
+                var folder = srcFolders.Key;
+                file.WriteLine($"### {folder}\n");
+
+                foreach (var subFolderTuple in srcFolders)
+                {
+                    var subfolder = subFolderTuple.Item1;
+                    var csproj = subFolderTuple.Item2;
 
-                file.WriteLine($"- [`{projectBullet}`]({link}): {desc}");
+                    var projectPath = $"{RepositoryRoot}\\{folder}\\{subfolder}\\{csproj}";
```

Comment:
Use `Path.DirectorySeparatorChar` when possible.

```suggestion
                    var s = Path.DirectorySeparatorChar;
                    var projectPath = $"{RepositoryRoot}{s}{folder}{s}{subfolder}{s}{csproj}";
```